encord/user_client.py

```diff
@@ -104,13 +108,25 @@ def get_or_create_dataset_api_key(self, dataset_hash: str) -> DatasetAPIKey:
         response = self.querier.basic_put(DatasetAPIKey, uid=None, payload=api_key_payload)
         return DatasetAPIKey.from_dict(response)
 
+    def get_datasets(self, properties_filter: Optional[Dict] = None) -> List[Dict]:
```

Comment:
Just checking, if the properties_filter parameter can be typed?

Something that can help the users build an instance without much effort

Reply:
It's a flat dictionary, and I don't really see how making is `Dict[Union[str, ListingFilter], Union[str, datetime.datetime]]` would be of much help to anyone. 

I'll look into maybe making the possible filters into **kwargs instead (and lose the enum below)

Reply:
Well it does tell people to look at the ListingFilter enum which is good for discoverability. I'd probably even drop the optional str and only take in ListingFilter. 

I'd also consider making the value of the dict to be some kind of 'Filter' interface so that people can discover the specific filter parameters in code

Reply:
good suggstion, made these in named parameters instead